backend/app/models/tables.py

```diff
@@ -0,0 +1,89 @@
+import enum
+from datetime import datetime, timezone
+from typing import Any
+
+from sqlalchemy import ARRAY, JSON, Column, String
+from sqlmodel import Enum, Field, Relationship, SQLModel
+
+
+class UserOrderLink(SQLModel, table=True):
+    user_id: int | None = Field(default=None, foreign_key="user.id", primary_key=True)
+    order_id: int | None = Field(default=None, foreign_key="order.id", primary_key=True)
+
+
+class UserProviderLink(SQLModel, table=True):
+    user_id: int | None = Field(default=None, foreign_key="user.id", primary_key=True)
+    provider_id: int | None = Field(default=None, foreign_key="provider.id", primary_key=True)
+
+
+class OrderProviderLink(SQLModel, table=True):
+    order_id: int | None = Field(default=None, foreign_key="order.id", primary_key=True)
+    provider_id: int | None = Field(default=None, foreign_key="provider.id", primary_key=True)
+
+
+class OrderStatus(str, enum.Enum):
+    SUBMITTED = "submitted"
+    ON_HOLD = "on_hold"
+    PROCESSING = "processing"
+    COMPLETED = "completed"
+    REJECTED = "rejected"
+    CANCELLED = "cancelled"
+
+
+class Order(SQLModel, table=True):
+    id: int | None = Field(default=None, primary_key=True)
+    created_at: datetime = datetime.now(tz=timezone.utc)
+    updated_at: datetime = datetime.now(tz=timezone.utc)
+    external_ref: str = Field(unique=True, nullable=False)
+    project_ref: str = Field(nullable=False)
```

Comment:
SQLmodel Field supports nullable? 
I couldnt find this param in SQLModel & pydantic doc. Its meant for SQLAlchemy Column right? I believe it works but its pretty complicated, which parameters are defined where.
I guess the explicit way will be to:
`project_ref: str = Field(sa_column=Column(String, nullable=False))`
 but I don't want to force anything cumbersome here. Maybe with some experience in SQLModel its obvious for others. Leaving up to you to decide